src/shims/intrinsics.rs

```diff
@@ -29,6 +27,22 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         // that might still hang around!
 
         let intrinsic_name = &*this.tcx.item_name(instance.def_id()).as_str();
+
+        // The 'abort' intrinsic has an empty 'dest', so we check for it before
+        // unwrapping it
+        if intrinsic_name == "abort" {
+            throw_ub_format!("the evaluated program aborted!");
```

Comment:
Aborting is not UB, though...

Reply:
As I discussed what @oli-obk, I wasn't really sure what error code to use here - we definitely want a stacktrace, though.

Reply:
Maybe extend the `Exit` error to also support a way to indicate "abnormal" program termination, and use that to show a stacktrace?

For now, you can also do `throw_unsup_format` with a FIXME.